native/src/main/java/io/ballerina/stdlib/kafka/nativeimpl/consumer/Poll.java

```diff
@@ -81,6 +83,30 @@ public static Object poll(Environment env, BObject consumerObject, BDecimal time
         return null;
     }
 
+    public static Object pollPayload(Environment env, BObject consumerObject, BDecimal timeout, BTypedesc bTypedesc) {
+        KafkaTracingUtil.traceResourceInvocation(env, consumerObject);
+        Future balFuture = env.markAsync();
+        KafkaConsumer kafkaConsumer = (KafkaConsumer) consumerObject.getNativeData(NATIVE_CONSUMER);
+        executorService.execute(()-> {
+            try {
+                Duration duration = Duration.ofMillis(getMilliSeconds(timeout));
+                ConsumerRecords recordsRetrieved = kafkaConsumer.poll(duration);
+                BArray dataArray = ValueCreator.createArrayValue((ArrayType) bTypedesc.getDescribingType());
```

Comment:
Since the typedesc is an array, we can change the `Type` - `ArrayType` along the flow rather casting it time to time in different places.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/474/commits/6bcfa1d491795102801bc37ae116043220b0f795